### PR TITLE
fixed dynamic  component registration

### PR DESCRIPTION
**Pull request by Revolyssup**
Signed-off-by: ashish <ashishjaitiwari15112000@gmail.com>

**Description**

This PR fixes dynamic component registration. Moves from helm based to manifest based.

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**

- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Revolyssup** [comment]
> I missed this. Why are we moving away from helm?

They moved their CRDs out of helm repo.

**Comment by leecalcote** [comment]
@Revolyssup is this ready: "You can add the "clone" traversal in meshkit and then use it here as well as consul adapter."?

**Comment by Revolyssup** [comment]
> @Revolyssup is this ready: "You can add the "clone" traversal in meshkit and then use it here as well as consul adapter."?

Can you make that meshkit release? I will push the final changes here

**Review by leecalcote — APPROVED** [review]
lgtm

**Review by leecalcote — CHANGES_REQUESTED** [review]
Build check failures, @Revolyssup  - https://github.com/meshery/meshery-osm/runs/4102911830?check_suite_focus=true

File: internal/config/releases.go [hunk 1]
```diff
@@ -107,3 +108,14 @@ func GetLatestReleases(releases uint) ([]*Release, error) {
 
 	return releaseList, nil
 }
+
+// GetFileNames takes the url of a github repo and the path to a directory. Then returns all the filenames from that directory
+func GetFileNames(owner string, repo string, path string) ([]string, error) {
+	g := walker.NewGit()
+	var filenames []string
+	err := g.Owner(owner).Repo(repo).Branch("master").Root(path).RegisterFileInterceptor(func(f walker.File) error {
+		filenames = append(filenames, f.Name)
```
Comment (utkarsh-pro) [inline internal/config/releases.go:117]: Assuming that this is concurrent in nature, won't this cause race issue?

**Review by utkarsh-pro — COMMENTED** [review]
Just one thing to address @Revolyssup

File: internal/config/releases.go (hunk 1 above)
Comment (Revolyssup) [reply internal/config/releases.go:117]: how does the concurrent nature can cause filenames array to be not what we expect by race condition?
Can you give an example?

Comment (utkarsh-pro) [reply internal/config/releases.go:117]: Slice after two appends => `[filename1][filename2][][]`
3 (or more) go routines try to append at the same time, all of them query for the capacity - all of them think they can add the value. Who wins?

**Comment by Revolyssup** [comment]
@utkarsh-pro  fixed it

**Comment by leecalcote** [comment]
@Revolyssup, will you take another look at the build failures, please?

File: internal/config/releases.go (hunk 1 above)
Comment (leecalcote) [reply internal/config/releases.go]: This is fixed, yeah?

**Review by leecalcote — APPROVED** [review]
Approving assuming that the workflow updates fix the failing static check.

**Review by utkarsh-pro — APPROVED** [review]
LGTM

**Comment by utkarsh-pro** [comment]
@Revolyssup can you check the failing check?

**Comment by leecalcote** [comment]
Issue #123 opened to track the failing static checks.

**Review by DelusionalOptimist — APPROVED** [review]
@Revolyssup all checks passing now. Merge whenever you're ready.

**Pull request merged**